tests/multi_step/test_correctness.py

```diff
@@ -22,9 +23,29 @@
     "16",
 ]
 
+NUM_LOGPROBS = [None, 5]  # `logprobs` argument to OpenAI completions API
 
-async def completions_with_server_args(prompts: List[str], model_name: str,
-                                       server_cli_args: List[str]):
+
+async def completions_with_server_args(
```

Comment:
We should have these utilities ready in conftest so no need to create duplicated ones.

Reply:
Moved them to tests/utils.py (did not see a clear benefit of using conftest vs utils.py)

Reply:
Sorry I didn't mean to move these utilities. I meant we already have something like `vllm_runner` in conftest that launches an engine for you to test. Please look at existing tests to make use of these utilities so that you don't need to create a new one.

Reply:
@comaniac ahh, yes I am very familiar with VllmRunner in conftest. However those existing conftest utilities do not serve the needed purpose.

When @SolitaryThinker wrote the multi-step test (test_correctness.py), he made it an end-to-end test. He starts up a separate server process, obtains a client for it, and sends completion requests to the OpenAI endpoint.

I think we do not have a pre-existing utility for this (probably why Will wrote it from scratch.) Certainly, looking through the other tests I have not found such a utility. That is why I moved Will's code to utils.py, so that it will become a utility.

VllmRunner is not an end-to-end test utility & I am pretty sure it relies on the sync API.